streams/src/main/java/org/apache/kafka/streams/kstream/Printed.java

```diff
@@ -0,0 +1,127 @@
+ *
+ * @param <K> key type
+ * @param <V> value type
+ * @see KStream#print(Printed)
+ */
+public class Printed<K, V> {
+    private final PrintWriter printWriter;
+    private String label;
+    private KeyValueMapper<? super K, ? super V, String> mapper = new KeyValueMapper<K, V, String>() {
+        @Override
+        public String apply(K key, V value) {
+            return String.format("%s, %s", key, value);
+        }
+    };
+
+    private Printed(final PrintWriter printWriter) {
+        this.printWriter = printWriter;
+    }
+
+    /**
+     * Builds the {@link ProcessorSupplier} that will be used to print the records flowing through a {@link KStream}
+     *
+     * @return the ProcessorSupplier to be used for printing
+     */
+    public ProcessorSupplier<K, V> build(final String processorName) {
+        return new KStreamPrint<>(new PrintForeachAction<>(printWriter, mapper, label != null ? label : processorName));
+    }
+
+    /**
+     * Print the records of a {@link KStream} to a file
+     *
+     * @param filePath path of the file
+     * @param <K>      key type
+     * @param <V>      value type
+     * @return a new Printed instance
+     */
+    public static <K, V> Printed<K, V> toFile(final String filePath) {
+        Objects.requireNonNull(filePath, "filePath can't be null");
+        if (filePath.trim().isEmpty()) {
+            throw new TopologyException("filePath can't be an empty string");
+        }
+        try {
+            return new Printed<>(new PrintWriter(filePath, StandardCharsets.UTF_8.name()));
+        } catch (FileNotFoundException | UnsupportedEncodingException e) {
+            throw new TopologyException("Unable to write stream to file at [" + filePath + "] " + e.getMessage());
+        }
+    }
+
+    /**
+     * Print the records of a {@link KStream} to system out
+     *
+     * @param <K> key type
+     * @param <V> value type
+     * @return a new Printed instance
+     */
+    public static <K, V> Printed<K, V> toSysOut() {
+        return new Printed<>(null);
+    }
+
+    /**
+     * Print the records of a {@link KStream} with the provided label
+     *
+     * @param label label to use
+     * @return this
+     */
+    public Printed<K, V> withLabel(final String label) {
+        Objects.requireNonNull(label, "label can't be null");
+        this.label = label;
+        return this;
+    }
+
+    /**
+     * Print the records of a {@link KStream} with the provided {@link KeyValueMapper}
+     * The provided KeyValueMapper's mapped value type must be {@code String}.
+     * <p>
+     * The example below shows how to customize output data.
+     * <pre>{@code
+     * final KeyValueMapper<Integer, String, String> mapper = new KeyValueMapper<Integer, String, String>() {
+     *     @Override
+     *     public String apply(Integer key, String value) {
+     *         return String.format("(%d, %s)", key, value);
+     *     }
+     * };
+     * }</pre>
+     * <p>
```

Comment:
nit: not need for `<p>` here as after the `<pre></per>` block there will be space inserted already